Makefile

```diff
@@ -11,51 +10,73 @@ GOBIN=$(shell go env GOBIN)
-	mkdir -p $@
-
 BRANCH ?= master
 # Paths to download the helm chart to.
-HELM_CHART_DEP_ROOT ?= $(BUILDDIR)/helmcharts
+HELM_CHART_DEP_ROOT = $(BUILDDIR)/helmcharts
 # Helm chart version and url
 HELM_CHART_VERSION ?= 24.4.1
 NGC_HELM_CHART_URL ?= https://helm.ngc.nvidia.com/nvidia/charts/network-operator-${HELM_CHART_VERSION}.tgz
 BRANCH_HELM_CHART_URL ?= https://github.com/Mellanox/network-operator/archive/refs/heads/${BRANCH}.tar.gz
+HELM_CHART_PATH ?=
+
+$(BUILDDIR) $(TOOLSDIR) $(HELM_CHART_DEP_ROOT): ; $(info Creating directory $@...)
+	mkdir -p $@
 
 # helm-docs is used to generate helm chart documentation
 HELM_DOCS_PKG := github.com/norwoodj/helm-docs/cmd/helm-docs
 HELM_DOCS_VER := v1.14.2
 HELM_DOCS_BIN := helm-docs
-HELM_DOCS_PATH = $(abspath $(TOOLSDIR)/$(HELM_DOCS_BIN))
 HELM_DOCS = $(abspath $(TOOLSDIR)/$(HELM_DOCS_BIN))-$(HELM_DOCS_VER)
 $(HELM_DOCS): | $(TOOLSDIR)
 	$(call go-install-tool,$(HELM_DOCS_PKG),$(HELM_DOCS_BIN),$(HELM_DOCS_VER))
-	$Q ln $(HELM_DOCS) $(HELM_DOCS_PATH)
 
 # go-install-tool will 'go install' a go module $1 with version $3 and install it with the name $2-$3 to $TOOLSDIR.
 define go-install-tool
-	$Q echo "Installing $(2)-$(3) to $(TOOLSDIR)"
-	$Q GOBIN=$(TOOLSDIR) go install $(1)@$(3)
-	$Q mv $(TOOLSDIR)/$(2) $(TOOLSDIR)/$(2)-$(3)
+	echo "Installing $(2)-$(3) to $(TOOLSDIR)"
+	GOBIN=$(TOOLSDIR) go install $(1)@$(3)
+	mv $(TOOLSDIR)/$(2) $(TOOLSDIR)/$(2)-$(3)
 endef
 
-download-ngc-helm-chart:
-	mkdir -p ${HELM_CHART_DEP_ROOT} && cd ${HELM_CHART_DEP_ROOT} \
+.PHONY: clean-helm-chart-dep-root
+clean-helm-chart-dep-root:
+	rm -rf ${HELM_CHART_DEP_ROOT}/*
+
+.PHONY: download-ngc-helm-chart
+download-ngc-helm-chart: | $(HELM_CHART_DEP_ROOT) clean-helm-chart-dep-root
+	cd ${HELM_CHART_DEP_ROOT} \
 	&& curl -sL ${NGC_HELM_CHART_URL} | tar -xz
 
-download-branch-helm-chart:
-	mkdir -p ${HELM_CHART_DEP_ROOT} \
-	&& curl -sL ${BRANCH_HELM_CHART_URL} \
+.PHONY: download-branch-helm-chart
+download-branch-helm-chart: | $(HELM_CHART_DEP_ROOT) clean-helm-chart-dep-root
+	curl -sL ${BRANCH_HELM_CHART_URL} \
 	| tar -xz -C ${HELM_CHART_DEP_ROOT} \
 	--strip-components 2 network-operator-${BRANCH}/deployment/network-operator
 
+.PHONY: copy-local-helm-chart
+copy-local-helm-chart: | $(HELM_CHART_DEP_ROOT) clean-helm-chart-dep-root
+	@if [ ! -d $(HELM_CHART_PATH) ] || [ -z $(HELM_CHART_PATH) ]; \
+		then echo "HELM_CHART_PATH is not a directory"; \
+		exit 1; \
+	fi
+	cp -r $(HELM_CHART_PATH) $(HELM_CHART_DEP_ROOT)
+
 # Generate helm chart documentation in a reStructuredText format.
-helm-docs: $(HELM_DOCS)
-	$(HELM_DOCS) --output-file=../../../../docs/customizations/helm.rst --ignore-file=.helmdocsignore --template-files=./templates/helm.rst.gotmpl ${HELM_CHART_DEP_ROOT}
+.PHONY: helm-docs
+helm-docs: | $(HELM_DOCS)
+	$(HELM_DOCS) --output-file=../../../../docs/customizations/helm.rst \
+	--ignore-file=.helmdocsignore \
+	--template-files=./templates/helm.rst.gotmpl \
+	--chart-search-root ${HELM_CHART_DEP_ROOT}
```

Comment:
i provided chart search root, since currently it was searching in the entire project, leading to warning when generating docs.

now it generates without warnings